Assets/Scripts/Puzzle.cs

```diff
@@ -39,16 +39,16 @@ public virtual string CurrentSolutionAttempt
             else
             {
                 currentSolutionAttempt_useProperty = "";
-                if (SolutionIsIncorrect != null)
-                    SolutionIsIncorrect.Invoke(this);
+                if (AttemptWasIncorrect != null)
+                    AttemptWasIncorrect.Invoke(this);
             }
         }
     }
 
     /// <summary>
     /// An event to invoke when the solution attempt fails
     /// </summary>
-    public static event System.Action<Puzzle> SolutionIsIncorrect;
+    public static event System.Action<Puzzle> AttemptWasIncorrect;
```

Comment:
Personally, I think this still sounds like a bool rather than an event. IncorrectSolutionAttempted? Just pick something that ends with an "ed" for me :)